Float enhance

wlbragg opened the pull request:
This can be merged anytime. It has a select animation so it only appears when floats or amphibious are selected.It also has the pontoon-uv stuff merged into it.

![fix2](https://cloud.githubusercontent.com/assets/11069680/10960951/f267a768-834e-11e5-9982-9480d552e815.jpg)

wlbragg commented:
The pontoon lighting effect changes are in the #511 PR. I also merged that into this branch,  why you might ask, I don't know, because I was getting to confused with what I did and didn't do. :confounded:

wlbragg commented:
OK, this is finished, or so I think. Please review and merge at your earliest convenience. 

One thing though, I put all the new Models/c172p.xml code a the end of the file instead of breaking it down and inserting it were it more appropriately belongs. I can do that before we merge if I get a "yes, we're going with this".

wlbragg commented:
Oops not quite, still needs flashlight effect applied.

gilbertohasnofb commented:
> Oops not quite, still needs flashlight effect applied.

Let me know when it's applied and I will test and merge this.

wlbragg commented:
Done

gilbertohasnofb commented:
Ok, I am testing it right now.

gilbertohasnofb commented:
Okay, here are some thoughts:

- I think you should bind the `g` key to the landing gear in `c172p-keyboard.xml`
- bug: the gear lever is being shown in all variations, see:
![fgfs-screen-001](https://cloud.githubusercontent.com/assets/5700795/10978840/643ac60e-83f9-11e5-8ec5-5daa3aa0522c.png)
- I think those two bars should stick to the panel and not the fuselage, more or less around the compass:
![tigert-photo](http://tigert.1g.fi/kuvat/Simulators/Flightgear/C172-dimensions/aan.jpg)

wlbragg commented:
>I think you should bind the g key to the landing gear in c172p-keyboard.xml

It is I think. "g" also controls the gear.

I'm working on the other now.

gilbertohasnofb commented:
> It is I think. "g" also controls the gear.

Really? It didn't work on my first test. As soon as you update this PR I will test it again, including the `g` shortcut.

wlbragg commented:
Pushed Gearswitch fix, still looking at the support bars.

wlbragg commented:
Yeah, your right on the support bars, that'll take a bit. It's not easy at my Blender expertise level.
I'm working on it now.

gilbertohasnofb commented:
Okay, so the keyboard binding worked for me now (I am having problems with the FailureManager, I think that it was the reason why it didn't work the first time). Also, the gear switch is fixed on all the other variants, thanks!

As for the editing the object in Blender, I guess you could just change the position a bit and that's all, just stick it right into the panel around the compass and that should be it for now :smile: 

But here is a "problem" I have: with the amphibian on a runway (and gears down, of course) as soon as I start moving it turns so much to the left, regardless of my rudder input. This also happens regardless of the water rudder position. Have you seen anything like this?

wlbragg commented:
I felt like I had to give it hard right the one time I tried it after putting in the instrument, how can that effect the steering? I thought it was just the way the original amphibious might have worked, maybe not. We need to look at the properties it's using to put the gear up and down and make sure they are the same as the default c172p "g" mapping is.

As far as the crossbars, I want them in there correct. My compass isn't showing and the GPS needs to be taken into account?

gilbertohasnofb commented:
> I felt like I had to give it hard right the one time I tried it after putting in the instrument, how can that effect the steering? I thought it was just the way the original amphibious might have worked, maybe not. We need to look at the properties it's using to put the gear up and down and make sure they are the same as the default c172p "g" mapping is.

I don't know, I just tried now twice: autostart and power on, and the plane goes crazily to the left even with full rudder to the right. I have not flown the amphibian almost at all so I don't know if this behaviour has always been there, or even if i am doing something stupid here!

> My compass isn't showing and the GPS needs to be taken into account?

Why your compass isn't showing?? As for the GPS, it's the GPS that should move, not the bars :wink: So when the GPS is ready, we can just position it somewhere else in the cockpit for the float and amphibious in case of collisions

wlbragg commented:
My compass is showing now, weird. OK, it's going to take a bit to get these bars where I want them. It may not be until late tonight.

I can switch branches and see it the amphibious pulls for a first test.

gilbertohasnofb commented:
> I can switch branches and see it the amphibious pulls for a first test.

I just tested with the `master` and the left pull is also there... :sob:

wlbragg commented:
It's master, not what I did.

wlbragg commented:
Yeah, I posted the same time as you.

gilbertohasnofb commented:
Okay, so should I open a new issue for that and then we try to solve it in this branch here?

wlbragg commented:
I guess a new issue would be best.

wlbragg commented:
I really don't care, I'm going to concentrate on getting what I started finished first.

gilbertohasnofb commented:
> I really don't care, I'm going to concentrate on getting what I started finished first.

Sure!

> I guess a new issue would be best.

Here it is: https://github.com/Juanvvc/c172p-detailed/issues/515

onox commented on Models/c172p.xml:
```diff
@@ -6955,4 +6888,252 @@
+        </action>
+    </animation>
+
+  <!-- TEXT OSG -->
```
Can you give this 4-space indentation like code above it?

wlbragg commented:
I think we are squared away on this.

model-combined-transparent and model-combined-deferred bot inherit model-combined and use the same cube mapping from what I could tell. I'm not sure where the differences are. I switched to deferred solely because I use another aircraft's example and that is what they were using. Transparent is giving us what we want and I think it is the correct one to use so I changed it.

OK, back to work. I am going to finish the support bracing and push again and I think we will be done here?

wlbragg commented:
@onox, I'm not sure with our compass positioned where it is the adding the cross piece will be visible at all from the inside. You will only see it from the outside, looking down from the top, close up.
Also note, we are moving it from its current location to right behind the compass.
Here are some un-textured images of the potential new placement. I am in the process of finalizing the position. It is difficult at best so I only want to do this once. Obviously I still need to move it back a little bit or down (which would give it even a bigger gap between the two supports), so it is not touching the compass. I have to quit for the night. @onox and @gilbertohasnofb, please review and advise where (precisely) you think we need to position the ends that are going into the dash and I will finish it up tomorrow late morning.

![fgfs-screen-007](https://cloud.githubusercontent.com/assets/11069680/10991158/83c7615e-8420-11e5-9333-919a944d9643.png)
![fgfs-screen-009](https://cloud.githubusercontent.com/assets/11069680/10991159/83c85dca-8420-11e5-99af-78806c131b00.png)
![fgfs-screen-006](https://cloud.githubusercontent.com/assets/11069680/10991157/83c6fae8-8420-11e5-89d4-bc827b019486.png)

wkitty42 commented:
FWIW: the pontoon craft both tried to go left just like the standard craft... the push was maybe slightly greater but not so much that one could not take off down the runway... remember that i flew the amphibious almost as soon as it appeared and if this steering problem had been apparent, i would have said something then...

onox commented:
@wlbragg It seems you got the compass fixed! :+1: 

There is one problem that you need to fix: you're using `gear/gear[4]/position-norm` up to `gear/gear[7]/position-norm`. But for the amphibious you need to use 19 up to 22 instead. 4 .. 7 are JSBsim contact points.

You can see that currently the gear lights in the cockpit don't show the correct state in the flight recorder. So just try to use 19 .. 22 in `Models/c172p.xml` and then replay mode should work.

onox commented:
@wlbragg Actually, make sure you use `alias="/params/gear-amphibious/front-left-position-norm"`, etc.

wlbragg commented:
Gotcha!

wlbragg commented:
Hopefully that is all it needs.

onox commented:
Let me check. Anything else that needs to be done or can I merge it after testing?

wlbragg commented:
I think I am done.

onox commented:
@wlbragg Which issues does this PR fix?

onox commented:
#507, #313, and #206?

wlbragg commented:
#313, #314, #206

Leave #507 open for now.

wlbragg commented:
#314 was addressed somewhere, I don't remember where.

#507, is back, I don't get it. It was working and now it's not. I need to make sure I got it fixed.

onox commented:
I prefer not to close #314 until we can get a strong reflection like in https://github.com/Juanvvc/c172p-detailed/issues/314#issuecomment-113593767.

wlbragg commented:
OK, I'll keep at it till I figure it out.

gilbertohasnofb commented:
@wlbragg Looks beautiful

wlbragg commented:
Thanks gilbertohasnofb, it's all in the texturing! :wink:
